torch/csrc/jit/python/python_ivalue.h

```diff
@@ -11,17 +11,21 @@ namespace ivalue {
 struct C10_EXPORT ConcretePyObjectHolder final : PyObjectHolder {
  public:
   static c10::intrusive_ptr<PyObjectHolder> create(py::object py_obj) {
-    return c10::make_intrusive<ConcretePyObjectHolder>(py_obj);
+    return c10::make_intrusive<ConcretePyObjectHolder>(std::move(py_obj));
```

Comment:
can we move the gil acquisition to the constructor as well (i.e. move the logic from toIValue to here)?

Reply:
I guess it's to change the constructor takes py::handle, and cast it to py::object with GIL

Reply:
Yes :)

Reply:
Actually, I would prefer adding a new constructor that takes a `py::handle`,  and move the logic to there. We should still keep the constructor that takes `py::object` without acquire gil for C++ usage.